packages/zoe/src/contractFacet/contractFacet.js

```diff
@@ -364,8 +364,16 @@ export function buildRootObject(_powers, _params, testJigSetter = undefined) {
         return invitationP;
       },
       // Shutdown the entire vat and give payouts
-      shutdown: () => {
-        E(zoeInstanceAdmin).shutdown();
+      shutdown: msg => {
+        E(zoeInstanceAdmin).terminate(msg);
```

Comment:
This appears to not be making a distinction between "step accepting offers" and "kill the vat".  An ongoing contract may stop the process of getting more offers. Also, what are the guarantees about `zcf.terminate` vs. turns and cranks?

Reply:
> This appears to not be making a distinction between "step accepting offers" and "kill the vat".

That's correct. There's recently added code in Zoe to ensure that when a vat is terminated zoe won't accept additional offers for that instance. If we wanted to, it would be straightforward to add a message from the contract to Zoe telling it to not accept additional offers. 

Should we add that to our task list? (I'm pretty sure that's orthogonal to this PR.)

>  An ongoing contract may stop the process of getting more offers. 

Not currently, but it's a reasonable feature to add.

> Also, what are the guarantees about zcf.terminate vs. turns and cranks?

`zcf.terminate()` is an eventual message from ZCF to Zoe, so other messages in the same turn within the ZCF vat will have an opportunity to finish what they're doing. When Zoe receives the message, it sends an eventual message to the vatAdminVat, , which talks to the vatAdmin device to shutdown the contract vat. From the vatAdminVat through the device to the kernel, the messages are synchronous. Until the kernel's `terminateVat()` runs, nothing prevents further turns in the contract vat, but nothing guarantees they'll get more turns, either. When the kernel runs, it cauterizes all the outstanding promises, and then resolves teh `done` promise, which is the last anyone hears from it.

I think the guarantees are
 * the code in the vat gets to complete the current turn
 * eventually, the code inside will be cut off from the outside world, and 
 * outstanding promises will be resolved or rejected.